### ISSUE TITLE
Use index section as resources

### ISSUE THREAD

**Issue by hayatoito** [WICG/webpackage#707]
As [Bundle Dependencies](https://github.com/hayatoito/webpackage/blob/dep/extensions/dependencies-section.md) proposal briefly mentions, it would be nice that a resource listed in the **index** section of the bundle will be also a target of subresource loading, even if the resource is not mentioned in `resources` in `<script type=webbundle>`.

The benefit is: users don't have to list all resources in `resources`. They have to specify only direct dependencies.

For example, suppose that `main.html` is:

```html
<script type="webbundle">
{
   source: "a.wbn",
   resources: [....]    # what should be specified?
}
</script>

<link rel="style" href="a.css" />

...

<img src="a.png" />

...

<script type="module" src="a1.js" />
```

The bundle, `a.wbn` has the following resources:

| URL     |
| ------- |
| ./a1.js |
| ./a2.js |
| ./a3.js |
| ./a.png |
| ./a.css |

The dependency graph is:

```
main.html
├── a1.js
│   ├── a2.js
│   └── a3.js
├── a.png
└── a.css
```

Then, it would be nice that users can specify only the direct dependencies in `resources`, as follows:

```html
<script type="webbundle">
{
   source: "a.wbn",
   resources: ["a1.js", "a.css", "a.png"]
}
</script>

<link rel="style" href="a.css" />

...

<img src="a.png" />

...

<script type="module" src="a1.js" />
```

Instead of listing all resources:

```html
<script type="webbundle">
{
   source: "a.wbn",
   resources: ["a1.js", "a2.js", "a3.js", "a.css", "a.png"]
}
</script>

<link rel="style" href="a.css" />

...

<img src="a.png" />

...

<script type="module" src="a1.js" />
```

We probably want to introduce an optional key in JSON of `<script type=webbundle>` to enable this behavior of not, however, I'm feeling that this behavior is fine as the default behavior in most cases .

**Comment by irori** [comment]
I vaguely remember that we considered using bundle's index section instead of letting the page specify the resources or scopes, although I cannot recall why we ended up with current design. The explainer has [No declarative scope](https://github.com/WICG/webpackage/blob/main/explainers/subresource-loading.md#no-declarative-scope) section for that idea.

**Comment by hayatoito** [comment]
Yeah, as far as I remember vaguely, the discussion in the past assumes "all" or "nothing".
Specify all resources or specify nothing.

This issue is proposing a slightly different thing. Specify only the direct dependencies beforehand. Regarding indirect dependencies, we can use *index* section information, in a theory.
